static/js/controllers/contacts-content.js

```diff
@@ -11,11 +11,12 @@
       $scope.selectContact($stateParams.id);
 
       $scope.$on('ContactUpdated', function(e, contact) {
-        if (!contact || (
-            contact._deleted &&
-            $scope.selected &&
-            $scope.selected._id === contact._id)) {
+        if (!contact) {
```

Comment:
If importing fired its own event we'd have to listen for that event in all the same places we listen for the current one. For me it makes much more sense to have one event and pass parameters to distinguish between the update types. Maybe we should change the param from a Contact to a list of Contacts, then this function can pass through all the contacts that were imported, and listeners who care can iterate through the list.